### PR TITLE
clutter is now stackable

### PR DESCRIPTION
**Pull request by soridanm**

### CODE REVIEW

**Comment by Piratkopia13** [comment]
Stuff can spawn with its center of mass outside whatever its on. Should be fixed
image

Clutter spawns in the way of doorways, its happening too often. Should be adjusted.
image

File: Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp [hunk 1]
```diff
@@ -1375,91 +1376,95 @@ void LevelGeneratorSystem::generateClutter() {
...
-		//		cS.height = 0.5f + cM.height;
-		//		cS.rot = rand() % 360;
-		//		map->smallClutter.push(cS);
-		//	}
-		//	map->mediumClutter.push(cM);
-		//}
+		//adds clutter on top of large objects
+		int amount = map->largeClutter.size();
+		for (int i = 0; i < amount; i++) {
+			clutter cL = map->largeClutter.front();
```
Comment (Piratkopia13) [inline Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp]: "clutter" enum is not following naming convention. Should be capitalised as "Clutter".

File: Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp [hunk 2]
```diff
@@ -1375,91 +1376,95 @@ void LevelGeneratorSystem::generateClutter() {
...
+			if (rand() % 100 < map->clutterModifier) {
+
+				//adds either one or two objects that are either medium or small
+				if(rand()%2==0){
+					//add on first half
+					{
+						float posx = (rand() % 40) / 100.f - 0.7f;
+						float posy = (rand() % 40) / 100.f - 0.2f;
+
+						clutter cM;
```
Comment (Piratkopia13) [inline Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp]: maybe rename cM to "clutterMedium" as it is easier to understand

File: Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp [hunk 3]
```diff
@@ -1375,91 +1376,95 @@ void LevelGeneratorSystem::generateClutter() {
...
+		//adds small clutter on top of medium clutter
+		amount = map->mediumClutter.size();
+		for (int i = 0; i < amount; i++) {
+			clutter cM = map->mediumClutter.front();
+			map->mediumClutter.pop();
+			if (rand() % 100 < map->clutterModifier) {
+				float posx = (rand() % 50) / 100.f - 0.25f;
+				float posy = (rand() % 50) / 100.f - 0.25f;
+
+				clutter cS;
```
Comment (Piratkopia13) [inline Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp]: rename

File: Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp (hunk 1 above)
Comment (soridanm) [reply Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp]: Also fixed rect to Rect

**Pull request merged**